src/App.tsx

```diff
@@ -1,26 +1,43 @@
-import React, { Component } from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import { FindMovie } from './components/FindMovie';
 import data from './api/movies.json';
+import { MoviesCard } from './components/interfaces';
 
-export class App extends Component {
-  state = {
-    movies: data,
+export const App: React.FC = () => {
+  const [hasAlready, setHasAlreadyStatus] = useState(false);
+  const [movies, setMovies] = useState(data);
+
+  const isNotHasAlready = () => {
+    setHasAlreadyStatus(false);
   };
 
-  render() {
-    const { movies } = this.state;
+  const addFilm = (newFilm: MoviesCard) => {
```

Comment:
```suggestion
  const addFilm = (movie: Movie) => {
```
дивись писати в параметрах функції addFilm що це новий фільм немає потреби, якщо ти "додаєш" якийсь фільм кудись, само собою що в параметрах логічно "новий" фільм, уточнювати це не треба

Reply:
Це з тої ж опери коли писати
```
const car = {
  carWheels: 4,
  carDoors: 5,
  carEngine: 1,
}
```
car всередині ключа додаткових даних про ключ не дає, само собою що двері в об'єкті car відносяться до машини, а не до будинку))

Reply:
Поменял на film, а не movie, потому что ниже movie в .some использую

Reply:
в таких ситуаціях currMovie можна юзати в колбеках